File: ruby/leetcode/RemoveElement_27.rb
# 27 Remove Element

# @param {Integer[]} nums
# @param {Integer} val
# @return {Integer}
def remove_element(nums, val)
  return 0 if nums.length == 0

  nums.each_with_index do |num, i|
    nums[i] = nil if num == val
  end
  nums.compact!

  nums.length
end


nums = [3,2,2,3]
val = 3
p remove_element(nums, val)

File: java/problem_solving/src/leetcode/RemoveElement_27.java
package leetcode;

// #27 - Remove Element

public class RemoveElement_27 {
	public static void main(String[] args) {
		int[] nums = {0, 1, 2, 2, 3, 0, 4, 2};
		int val = 2;

		System.out.println("val = " + removeElement(nums, val));
	}

	public static int removeElement(int[] nums, int val) {
		int idx = 0;
		for(int i=0; i<nums.length; i++){
			if(nums[i] != val){
				nums[idx++] = nums[i];
			}
		}
		return idx;

	}
}
